examples/nlp/fnet_classification_with_keras_hub.py

```diff
@@ -213,26 +242,89 @@ def train_word_piece(ds, vocab_size, reserved_tokens):
 
 
 """
-## Formatting the dataset
-
-Next, we'll format our datasets in the form that will be fed to the models. We
-need to tokenize the text.
-"""
-
-
-def format_dataset(sentence, label):
-    sentence = tokenizer(sentence)
-    return ({"input_ids": sentence}, label)
-
-
-def make_dataset(dataset):
-    dataset = dataset.map(format_dataset, num_parallel_calls=tf.data.AUTOTUNE)
-    return dataset.shuffle(512).prefetch(16).cache()
+Create the final datasets, method adapted from `PyDataset` doc string
+"""
+
+
+class UnifiedPyDataset(keras.utils.PyDataset):
```

Comment:
Something similar to this already exists in Keras. Keras can natively consume Pandas DataFrames (now I'm realizing the documentation doesn't even mention it). You just pass DataFrames as the `x` and `y` parameters to `model.fit()`.